packages/roc-package-web-app-react/app/server/router.js

```diff
@@ -33,7 +34,10 @@ export default function routes({ createRoutes, createStore, stats, dist }) {
                 this.status = 200;
                 this.body = renderPage();
             } else {
-                const store = createStore ? createStore() : null;
+                const { store, history, url } = setupForRender(createStore, this.url);
+
+                // Give Koa middlewares a chance to interact with the reduxStore
+                // This can be used to dynamically pass some data to the client.
```

Comment:
I do not thing so unfortunately, this because of two reasons. I hope that I'm missing something here since this would be a much better "API".

Firstly we need to do `syncHistoryWithStore` before we use the `history` instance in `match`. Maybe this can be avoided if we instead manually update the store to match the whatever state we get with this method.

Secondly we are currently passing in the `store` to the function that creates the routes. I think we want this to be able to use things like `onEnter` to check if the user is authenticated and if not perform a redirect.